docs/adding_a_new_gravity_endpoint_into_metaphysics.md

```diff
@@ -0,0 +1,78 @@
+## Adding a gravity endpoint into Metaphysics
+
+This doc is aimed at helping backend devs or beginner front end devs on how to add an existing gravity endpoint into Metaphysics.
+
+This article assumes that you already have your local machine setup and have your graphiql.app or other app (Postman, Insomnia etc) setup. If you need help with that, you can start [here](https://github.com/artsy/metaphysics#setting-up-your-local-graphiql)
+
+### Steps to add an existing endpoint
+
+1. Find an existing PR or commit which adds something similar to what you want. If you are looking for a PUT endpoint, then [this](https://github.com/artsy/metaphysics/commit/987b21f501bb3483d196720d4a252c25b704352e) is a great example.
+1. Follow the structure of the PR to add your own endpoint and input/outputs. Some further tips can be found below.
+1. After the code has been written, you can run `yarn dump:local` in the Metaphysics directory to generate the new schema files. This script (and other useful scripts) can be found in the `package.json` file.
+1. Start your local server with `yarn start`
+1. With your favourite graphql app, you should be able to hit the local server to refresh the schema. In insomnia, you can click on the "schema" button in the GraphQL panel to manually refresh this, if you don't have auto-refresh turned on (also found in this list)
+1. You should be able to hit your local metaphysics, which will hit staging gravity by default.
+1. Work through the errors, making sure to read the message in its entirety. Also don't hesitate to add it to the errors section below if useful.
+
+### Tips on writing and debugging locally
+
+- GraphQL is strongly typed and needs to know the exact inputs and outputs of the endpoint as it mediates between the clients and the REST endpoint. This is why you need to set up a union type which instructs GraphQL on what both responses will look like. This allows GraphQL tools to know if the request you are making (in Insomnia, for example) is valid, even before you send the request.
```

Comment:
Going from setup to writing mutation-handling Union types is a very steep curve! 

As we want this document to be instructive (rather than intimidating) its best to keep this all very incremental versus skipping ahead. Meaning: 

- Add an example that shows how to add a new loader pointing at a Gravity `GET` endpoint 
- Show how to add a new field to a section within the schema (say, `artwork.ts`) 
- Show how to add a new `POST` endpoint in prep for a mutation
- Show had to add a new mutation (in the simplest way possible; no fancy error handling is necessary yet; see all of the mutations in the `me` folder for an example)
- THEN show how to handle errors in a way that is more sophisticated. 

But even so, this `thingOrError` pattern isn't necessarily something we should advocate for up front in a getting started doc. In many ways it's overly complicated for everyday mutations and should be used more on a need to know basis. 

(Worth noting that we have a full blog post on this here: https://artsy.github.io/blog/2018/10/19/where-art-thou-my-error/)

Reply:
Thanks for the article link, I will have to check it out and also link it in this doc.

I guess I'm simultaneously both the best and worst person to make this doc. The best person because I know almost nothing about GraphQL and can give feedback on whether something is too complicated or not, and also the worst person because I know nothing about GraphQL and so I shouldn't be teaching others about things I don't really understand.

The `thingOrError` I thought was standard and came from the [example PR](https://github.com/artsy/metaphysics/pull/3918) I was basing my changes on. Happy to learn the easier way on how to do this!